apps/e2e/cypress/e2e/talentsearch/submit-application-iap.cy.js

```diff
@@ -0,0 +1,299 @@
+        currentCity: "Test City",
+        telephone: "+10123456789",
+        armedForcesStatus: ArmedForcesStatus.NonCaf,
+        citizenship: CitizenshipStatus.Citizen,
+        lookingForEnglish: true,
+        isGovEmployee: false,
+        hasPriorityEntitlement: false,
+        locationPreferences: WorkRegion.Ontario,
+        positionDuration: PositionDuration.Permanent,
+        expectedGenericJobTitles: {
+          sync: testGenericJobTitleIds,
+        },
+      }).as("testUser");
+    });
+
+    cy.get("@testUser").then((testUser) => {
+      addRolesToUser(testUser.id, ["guest", "base_user", "applicant"]);
+    });
+
+    cy.getDCM().then((dcmId) => {
+      cy.getMe()
+        .its("id")
+        .then((adminUserId) => {
+          cy.get("@testClassificationId").then((testClassificationId) => {
+            cy.createPool(adminUserId, dcmId, [testClassificationId])
+              .its("id")
+              .as("testPoolId")
+              .then((testPoolId) => {
+                cy.get("@testSkillIds").then((testSkillIds) => {
+                  cy.updatePool(testPoolId, {
+                    name: {
+                      en: "Cypress Test Pool EN",
+                      fr: "Cypress Test Pool FR",
+                    },
+                    stream: PoolStream.BusinessAdvisoryServices,
+                    closingDate: `${FAR_FUTURE_DATE} 00:00:00`,
+                    yourImpact: { en: "test impact EN", fr: "test impact FR" },
+                    keyTasks: { en: "key task EN", fr: "key task FR" },
+                    essentialSkills: {
+                      sync: testSkillIds,
+                    },
+                    language: PoolLanguage.Various,
+                    securityClearance: SecurityStatus.Secret,
+                    location: {
+                      en: "test location EN",
+                      fr: "test location FR",
+                    },
+                    isRemote: true,
+                    publishingGroup: PublishingGroup.Iap,
+                  });
+                  cy.publishPool(testPoolId);
+                });
+              });
+          });
+        });
+    });
+
+    cy.get("@testUser").then((user) => {
+      cy.loginBySubject(user.sub);
+    });
+    cy.visit("/en/indigenous-it-apprentice");
+
+    // IAP home page
+    cy.findByRole("heading", {
+      name: /IT Apprenticeship Program for Indigenous Peoples/i,
+      level: 1,
+    })
+      .should("exist")
+      .and("be.visible");
+    cy.get("@testPoolId").then((testPoolId) => {
+      cy.get(
+        `a[href*="/browse/pools/${testPoolId}/create-application?personality=iap"]`,
+      )
+        .first() // there are several buttons on the page to "Apply Now"
+        .click();
+    });
+
+    // Welcome page - step one
+    cy.findByRole("heading", {
+      name: "Apply to Cypress Test Pool EN",
+      level: 1,
+    })
+      .should("exist")
+      .and("be.visible"); // IAP heading does not include job stream
```

Comment:
We may want to wait on the create application mutation before this. Mine is checking before the page renders 😬

Reply:
OK, adding waits back in.
Ref: [add query and mutation waits](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7096/commits/e4c6eb778853db4d481ba91d3adcdbbf1b702e5d)